### PR TITLE
ALTAPPS-1003: Shared rearrange home screen

### PR DESCRIPTION
**Pull request by vladkash**
**YouTrack Issues**:
[#ALTAPPS-1003](https://vyahhi.myjetbrains.com/youtrack/issue/ALTAPPS-1003)

**Checklist**

_Before Code Review:_

- [x] Fields "Assignees, Labels, Milestone" are filled in the pull request;
- [ ] All checks have been passed;
- [x] Changes have been checked locally.

**Description**
- Removed state for problems limit widget & next learning activity from home feature state
- Removed next learning activity widget module
- Replaced navigation from Home screen to Study plan (now study plan is main screen of the app)
- Removed mentions of "home" from multiplatform strings

### CODE REVIEW

**Comment by vladkash** [comment]
@XanderZhu @ivan-magda 
In this PR I didn't made changes on platforms for fixing CI because I think that compiler errors will help in platform tasks in order to don't forget something

**Comment by ivan-magda** [comment]
@vladkash we can modify the branch to merge from `feature/first_problem_onboarding` to `feature/rearrange_home_screen`.

`feature/first_problem_onboarding` has a complete navigation flow, no TODOs left on both platforms

**Review by XanderZhu — APPROVED** [review]

File: shared/src/commonMain/kotlin/org/hyperskill/app/home/presentation/HomeFeature.kt [hunk 1]
```diff
@@ -41,14 +34,12 @@ interface HomeFeature {
...
          * @property repetitionsState Topics repetitions state.
          * @property isRefreshing A boolean flag that indicates about is pull-to-refresh is ongoing.
-         * @property isLoadingMagicLink A boolean flag that indicates about magic link loading.
          * @see Streak
          * @see ProblemOfDayState
          */
         data class Content(
             val problemOfDayState: ProblemOfDayState,
             val repetitionsState: RepetitionsState,
-            val isFreemiumEnabled: Boolean,
```
Comment (ivan-magda) [inline shared/src/commonMain/kotlin/org/hyperskill/app/home/presentation/HomeFeature.kt:51]: `isFreemiumEnabled` We are using this flag on platforms for displaying "Solve unlimited" badge on the problem of day card

Comment (ivan-magda) [reply shared/src/commonMain/kotlin/org/hyperskill/app/home/presentation/HomeFeature.kt:51]: Now we have a bug:

```kotlin
is Message.HomeSuccess ->
    state.copy(
        homeState = HomeState.Content(
            message.problemOfDayState, // problemOfDayState
            message.repetitionsState, // repetitionsState
            message.isFreemiumEnabled // isRefreshing
        )
    ) to emptySet()
```

Comment (ivan-magda) [reply shared/src/commonMain/kotlin/org/hyperskill/app/home/presentation/HomeFeature.kt:51]: https://github.com/hyperskill/mobile-app/pull/707/commits/efb782cb052206e3aa4247d5d022a80097a1164b

Comment (ivan-magda) [reply shared/src/commonMain/kotlin/org/hyperskill/app/home/presentation/HomeFeature.kt:51]: cc @vladkash @XanderZhu

**Pull request merged**